src/main/java/org/jenkinsci/plugins/nomad/NomadApi.java

```diff
@@ -55,20 +54,19 @@ String checkResponseAndGetBody (Request request) {
         return bodyString;
     }
 
-    void startWorker(NomadCloud cloud, String workerName, String nomadToken, String jnlpSecret, NomadWorkerTemplate template) {
+    void startWorker(String workerName, String nomadToken, String jnlpSecret, NomadWorkerTemplate template) {
 
         String workerJob = buildWorkerJob(
                 workerName,
                 jnlpSecret,
-                cloud,
-                template
+                template.getJobTemplate()
         );
 
         LOGGER.log(Level.FINE, workerJob);
 
-        RequestBody body = RequestBody.create(JSON, workerJob);
+        RequestBody body = RequestBody.create(workerJob, JSON);
         Request.Builder builder = new Request.Builder()
-                .url(this.nomadApi + "/v1/job/" + workerName + "?region=" + template.getRegion());
+                .url(this.nomadApi + "/v1/jobs");
```

Comment:
Since `region` and `workerName` are part of the job file already, we don't have to define them here explicitly.